spec/migrations/20190617191109_move_awx_credentials_to_authentications_spec.rb

```diff
@@ -0,0 +1,209 @@
+      end
+
+      it "sets options to nil if there are none we care about" do
+        auth = authentication.create!(
+          :name        => "machine",
+          :userid      => "machine",
+          :type        => "ManageIQ::Providers::EmbeddedAnsible::AutomationManager::MachineCredential",
+          :manager_ref => "3",
+          :options     => {
+            :ssh_key_data    => "$encrypted$",
+            :become_username => "machine",
+            :become_password => "$encrypted$"
+          }
+        )
+
+        stub_awx_cred_for_id(awx_conn, "3", [])
+
+        migrate
+
+        expect(auth.reload.options).to be_nil
+      end
+
+      it "migrates options fields from the awx database" do
+        auth = authentication.create!(
+          :name        => "machine",
+          :userid      => "machine",
+          :type        => "ManageIQ::Providers::EmbeddedAnsible::AutomationManager::MachineCredential",
+          :manager_ref => "3",
+          :options     => {
+            :ssh_key_data => "$encrypted$"
+          }
+        )
+
+        awx_credential = {
+          "inputs" => {
+            "authorize"     => false,
+            "become_method" => "sudo",
+            "client"        => "someclient",
+            "domain"        => "thebestdomain",
+            "host"          => "host.example.com",
+            "project"       => "manageiq",
+            "subscription"  => "sub",
+            "tenant"        => "atenant"
+          }.to_json
+        }
+
+        stub_awx_cred_for_id(awx_conn, "3", [awx_credential])
+
+        migrate
+        auth.reload
+
+        expected_options = {
+          :authorize     => false,
+          :become_method => "sudo",
+          :client        => "someclient",
+          :domain        => "thebestdomain",
+          :host          => "host.example.com",
+          :project       => "manageiq",
+          :subscription  => "sub",
+          :tenant        => "atenant"
+        }
+
+        expect(auth.options).to eq(expected_options)
+      end
+
+      it "migrates real data" do
+        # load the initial set of authentication records
+        auths = YAML.load_file(data_dir.join("initial_vmdb_authentications.yaml"))
+        auths.each { |attrs| authentication.create!(attrs) }
+
+        stub_awx_credentials(awx_conn)
+        expect_credential_decrypts
+
+        migrate
+
+        expect_authentications_migrated
+      end
+    end
+  end
+
+  def stub_awx_cred_for_id(connection, id, data)
+    expect(connection).to receive(:async_exec).with("SELECT inputs FROM main_credential WHERE id = $1::BIGINT", [id]).and_return(data)
+  end
+
+  def stub_awx_credentials(connection)
+    credentials = YAML.load_file(data_dir.join("credential_attributes.yaml"))
+    credentials.each do |cred|
+      data = cred["awx_attributes"].to_json
+      stub_awx_cred_for_id(connection, cred["id"], [{"inputs" => data}])
+    end
+  end
+
+  def expect_credential_decrypts
```

Comment:
I guess I more have a concern about this from a higher level than @Fryguy 's concern below:

Is there a reason we are even bothering testing that `AwesomeSpawn` is receiving the proper arguments?  Isn't that just an implementation detail, and as long as stuff doesn't blow up, there is no point in testing it.  Specifically if we are potentially going to switch over to a ruby solution (if time permits).

Personally, I would rather see that the `python` script is being exercised, and that we receive the expected input and output from the migration.

Reply:
Actually, I might be able to answer this myself:  My guess is we don't have `django`/`ansible` on CI, so the `python` script is probably never exercised on travis.

The above comment probably can be ignored, since I really don't have a good alternative for you on that one... 😞

Reply:
So I can write up some specs that will skip if we don't have the dependencies we need.

If most people will run embedded ansible with runner locally than it's possible that the specs will run in a fair number of cases.

I can also look into installing what we need on CI ...